src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventPullerService.cs

```
@@ -35,9 +35,21 @@ public EventPullerService(
             .ToList();
     }
 
-    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        return Task.WhenAll(this._eventGridSubscriptionPullers.Select(puller => Task.Run(() => puller.StartReceivingEventsAsync(stoppingToken), stoppingToken)));
+        var subscriptionTasks = this._eventGridSubscriptionPullers
+            .Select(puller => Task.Run(() => puller.StartReceivingEventsAsync(stoppingToken), stoppingToken))
+            .ToList();
+
+        while (subscriptionTasks.Any())
```

We don't necessarily have to change to this but just as a fyi, here's something Antho sent 
https://stackoverflow.com/questions/57313252/how-can-i-await-an-array-of-tasks-and-stop-waiting-on-first-exception

---

It's a very interesting way to optimize handling tasks and failing fast!

However, in our case, once the tasks are started they should be "eternal", so performance gains in handling multiple tasks are less important. Overall, here I feel like this would be a more complex logic than what is suggested, for limited gains.

I'm still going to keep that in mind for future use cases, though!

---

Also, I don't think perf is important here. In best case scenario, the tasks never complete.